packages/bot-web-ui/src/stores/quick-strategy-store.js

```diff
@@ -231,20 +237,24 @@ export default class QuickStrategyStore {
             }
         });
 
-        const file_name = strategies[strategy_name].label;
-        const workspace = Blockly.derivWorkspace;
+        const file_name = strategies?.[strategy_name]?.label || localize('Unknown');
+        const { derivWorkspace: workspace } = Blockly;
+
         load({ block_string: Blockly.Xml.domToText(strategy_dom), file_name, workspace, from: save_types.UNSAVED });
 
         if (button === 'run') {
-            const trade_definition_block = workspace.getTradeDefinitionBlock();
-
-            workspace.waitForBlockEvent(trade_definition_block.id, Blockly.Events.BLOCK_CREATE).then(() => {
-                this.toggleStrategyModal();
-                this.root_store.run_panel.onRunButtonClick();
-            });
-        } else {
+            await workspace.waitForBlockEvent(workspace.getTradeDefinitionBlock().id, Blockly.Events.BLOCK_CREATE);
+        }
+        // Close modal if open for edit + run.
+        if (this.is_strategy_modal_open) {
```

Comment:
Should this be under the `if (button === 'run')` condition if we want to close the modal if it's open for edit and then run.

Reply:
In both cases (Create & edit + Run) we want to toggle the modal, so putting it under `if (button === 'run') would not be correct.